x-pack/plugins/observability/public/hooks/use_time_range.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TimePickerTime } from '../components/shared/data_picker';
+import { getAbsoluteTime } from '../utils/date';
+import { useKibanaUISettings, UI_SETTINGS } from './use_kibana_ui_settings';
+import { usePluginContext } from './use_plugin_context';
+
+export function useTimeRange({ rangeFrom, rangeTo }: { rangeFrom?: string; rangeTo?: string }) {
+  const { plugins } = usePluginContext();
+  const timePickerTimeDefaults = useKibanaUISettings<TimePickerTime>(
+    UI_SETTINGS.TIMEPICKER_TIME_DEFAULTS
+  );
+
+  const timePickerSharedState = plugins.data.query.timefilter.timefilter.getTime();
+
+  const _rangeFrom = rangeFrom ?? timePickerSharedState.from ?? timePickerTimeDefaults.from;
+  const _rangeTo = rangeTo ?? timePickerSharedState.to ?? timePickerTimeDefaults.to;
+
+  return {
+    rangeFrom: _rangeFrom,
+    rangeTo: _rangeTo,
+    absStart: getAbsoluteTime(_rangeFrom)!,
+    absEnd: getAbsoluteTime(_rangeTo, { roundUp: true })!,
+  };
+}
```

Comment:
Why does this function take `rangeFrom` and `rangeTo` as input? Are those defaults?

Reply:
These are values that might come in the query string. For instance, the Overview page might have it already defined, if not uses the default ones.

Reply:
Any reason why we can't handle that in this hook as well?

Reply:
Basically `useTimeRange` is a generic hook, when I say generic I mean that it can be called anywhere, any page. `useRouteParams` receives a pathname as an argument, I think it would be weird to call it inside  `useTimeRange` passing `/overview`. WDYT?

Reply:
Ah, that makes sense then 👍